src/datahike/query.cljc

```diff
@@ -48,11 +49,49 @@
 ;; or [ (Datom. 2 "Oleg" 1 55) ... ]
 (defrecord Relation [attrs tuples])
 
+;; Main functions
+
+(defn normalize-q-input
```

Comment:
Would be good to have a unit test for this function.

Reply:
Started doing this, seems I am having some trouble with my local setup atm though, so I have to solve this first.

Reply:
The tests are finished now.